src/wifi.rs

```diff
@@ -1355,3 +1358,110 @@ impl WifiWait {
         }
     }
 }
+
+#[cfg(all(feature = "nightly", feature = "experimental"))]
+pub struct EspAsyncWifiDriver<'d> {
+    sync: WifiDriver<'d>,
+}
+
+#[cfg(all(feature = "nightly", feature = "experimental"))]
+impl<'d> EspAsyncWifiDriver<'d> {
+    #[cfg(all(feature = "alloc", esp_idf_comp_nvs_flash_enabled))]
+    pub fn new<M: WifiModemPeripheral>(
+        modem: impl Peripheral<P = M> + 'd,
+        sysloop: EspSystemEventLoop,
+        nvs: Option<EspDefaultNvsPartition>,
+    ) -> Result<Self, EspError> {
+        Ok(EspAsyncWifiDriver {
+            sync: WifiDriver::new(modem, sysloop, nvs)?,
+        })
+    }
+}
+
+#[cfg(all(feature = "nightly", feature = "experimental"))]
+impl<'d> asynch::Wifi for EspAsyncWifiDriver<'d> {
```

Comment:
The current philosophy of the esp-idf-hal crate is that everything accessible via the e-hal traits should also be doable via an esp-idf-hal API which is straight in the driver struct. This way you don't need nightly experimental features if you don't like these. Therefore, please move your stubs to the driver (as regular async methods) and delegate to these from the trait impl.